content/en/blog/_posts/2022-XX-XX-iptables-chains.md

```diff
@@ -0,0 +1,151 @@
+as a reminder to itself. Then later, during `POSTROUTING`, it has a
+single rule that matches all previously-marked packets, and calls `-j
+MASQUERADE` on them.
+
+If you have _a lot_ of rules where you need to apply both DNAT and
+masquerading to the same packets like kube-proxy does, then you may
+want a similar arrangement. But in many cases, components that use
+`KUBE-MARK-MASQ` are only doing it because they copied kube-proxy’s
+behavior without understanding why kube-proxy was doing it that way.
+Many of these components could easily be rewritten to just use
+separate DNAT and masquerade rules. (In cases where no DNAT is
+occurring then there is even less point to using `KUBE-MARK-MASQ`;
+just move your rules from `PREROUTING` to `POSTROUTING` and call `-j
+MASQUERADE` directly.)
+
+### If you use `KUBE-MARK-DROP`... {#use-case-kube-mark-drop}
+
+The rationale for `KUBE-MARK-DROP` is similar to the rationale for
+`KUBE-MARK-MASQ`: kube-proxy wanted to make packet-dropping decisions
+alongside other decisions in the `nat` `KUBE-SERVICES` chain, but you
+can only call `-j DROP` from the `filter` table. So instead, it uses
+`KUBE-MARK-DROP` to mark packets to be dropped later on.
+
+In general, the approach for removing a dependency on `KUBE-MARK-DROP`
+is the same as for removing a dependency on `KUBE-MARK-MASQ`. In
+kube-proxy’s case, it is actually quite easy to replace the usage of
+`KUBE-MARK-DROP` in the `nat` table with direct calls to `DROP` in the
+`filter` table, because there are no complicated interactions between
+DNAT rules and drop rules, and so the drop rules can simply be moved
+from `nat` to `filter`.
+
+In more complicated cases, it might be necessary to “re-match” the
+same packets in both `nat` and `filter`.
+
+### If you use Kubelet’s IPTables rules to figure out `iptables-legacy` vs `iptables-nft`... {#use-case-iptables-mode}
+
+Components that manipulate host-network-namespace iptables rules from
+inside a container need some way to figure out whether the host is
+using the old `iptables-legacy` binaries or the newer `iptables-nft`
+binaries (which talk to a different kernel API underneath).
+
+The [`iptables-wrappers`] module provides a way for such components to
+autodetect the system iptables mode, but in the past it did this by
+assuming that Kubelet will have created “a bunch” of iptables rules
+before any containers start, and so it can guess which mode the
+iptables binaries in the host filesystem are using by seeing which
+mode has more rules defined.
+
+In future releases, Kubelet will no longer create many iptables rules,
+so heuristics based on counting the number of rules present may fail.
+
+However, starting in 1.24, Kubelet will always create a chain named
+`KUBE-IPTABLES-HINT` in the `mangle` table of whichever iptables
+subsystem it is using. Components can now look for this specific chain
+to know which iptables subsystem Kubelet (and thus, presumably, the
+rest of the system) is using.
```

Comment:
v1.24 is already released; did you mean v1.25? Or maybe you're explaining an existing detail? (if so, we might be able to link to the release note heading about this).

Reply:
It's in 1.24 but I wrote the text before 1.24 was released. Probably should say "as of 1.24" or something...